test/Microsoft.TemplateEngine.TestHelper/EnvironmentSettingsHelper.cs

```
@@ -0,0 +1,67 @@
+    {
+        List<string> foldersToCleanup = new List<string>();
+        List<EngineEnvironmentSettings> engineEnvironmentToDispose = new List<EngineEnvironmentSettings>();
+
+        public IEngineEnvironmentSettings CreateEnvironment(string locale = "en-US")
+        {
+            Environment.SetEnvironmentVariable(RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "USERPROFILE" : "HOME", CreateTemporaryFolder());
+            ITemplateEngineHost host = new TestHost
+            {
+                HostIdentifier = "TestRunner",
+                Version = "1.0.0.0",
+                Locale = locale,
+                BuiltInComponents = new AssemblyComponentCatalog(new List<Assembly>()
+                {
+                    typeof(RunnableProjectGenerator).Assembly,//RunnableProject
+                    typeof(SettingsLoader).Assembly//Edge
+                }),
+                FileSystem = new MonitoredFileSystem(new PhysicalFileSystem()),
+                FallbackHostTemplateConfigNames = new[] { "dotnetcli" }
+            };
+
+            var engineEnvironmentSettings = new EngineEnvironmentSettings(host, (x) => new SettingsLoader(x));
+            engineEnvironmentToDispose.Add(engineEnvironmentSettings);
+            return engineEnvironmentSettings;
+        }
+
+        public string CreateTemporaryFolder()
+        {
+            var folder = Path.Combine(Path.GetTempPath(), "DotnetNew3_Tests", Guid.NewGuid().ToString(), nameof(EnvironmentSettingsHelper));
+            foldersToCleanup.Add(folder);
+            Directory.CreateDirectory(folder);
+            return folder;
+        }
+
+        public void Dispose()
+        {
+            try
+            {
+                engineEnvironmentToDispose.ForEach(e => e.Dispose());
+                foldersToCleanup.ForEach(f => Directory.Delete(f, true));
+            }
+            catch (Exception)
+            {
+                //Sometimes randomly fails :(
```

please persist the exception details to understand more the failures instead of swallowing the exception.

---

I removed this catching and after running few times doesn't seem to be failing anymore.